mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+    // Creating the file containing the data files generated by this task
+    createFileForCommit(closedFiles, fileForCommitLocation, io(context.getJobConf()));
+  }
+
+  /**
+   * Removes files generated by this task.
+   * @param context The task context
+   * @throws IOException Thrown if there is an error closing the writer
+   */
+  @Override
+  public void abortTask(TaskAttemptContext context) throws IOException {
+    // Clean up writer data from the local store
+    HiveIcebergRecordWriter writer = HiveIcebergRecordWriter.removeWriter(context.getTaskAttemptID());
+
+    // Remove files if it was not done already
+    writer.close(true);
+  }
+
+  /**
+   * Reads the commit files stored in the temp directory and collects the generated committed data files.
+   * Appends the data files to the table. At the end removes the temporary directory.
+   * @param jobContext The job context
+   */
+  @Override
+  public void commitJob(JobContext jobContext) {
+    JobConf conf = jobContext.getJobConf();
+    Table table = Catalogs.loadTable(conf);
+
+    long startTime = System.currentTimeMillis();
+    LOG.info("Committing job has started for table: {}, using location: {}", table,
+        LocationHelper.generateJobLocation(conf, jobContext.getJobID()));
+
+    List<DataFile> dataFiles = dataFiles(jobContext, io(jobContext.getJobConf()), true);
+
+    if (dataFiles.size() > 0) {
+      // Appending data files to the table
+      AppendFiles append = table.newAppend();
+      dataFiles.forEach(append::appendFile);
+      append.commit();
+      LOG.info("Commit took {} ms for table: {} with {} file(s)", System.currentTimeMillis() - startTime, table,
+          dataFiles.size());
+      LOG.debug("Added files {}", dataFiles);
+    } else {
+      LOG.info("Commit took {} ms for table: {} with no new files", System.currentTimeMillis() - startTime, table);
+    }
+
+    cleanup(jobContext);
+  }
+
+  /**
+   * Removes the generated data files, if there is a commit file already generated for them.
+   * The cleanup at the end removes the temporary directory as well.
+   * @param jobContext The job context
+   * @param status The status of the job
+   */
+  @Override
+  public void abortJob(JobContext jobContext, int status) {
+    String location = LocationHelper.generateJobLocation(jobContext.getJobConf(), jobContext.getJobID());
+    LOG.info("Job {} is aborted. Cleaning job location {}", jobContext.getJobID(), location);
+
+    List<DataFile> dataFiles = dataFiles(jobContext, io(jobContext.getJobConf()), false);
+
+    // Check if we have files already committed and remove data files if there are any
+    if (dataFiles.size() > 0) {
+      Tasks.foreach(dataFiles)
+          .retry(3)
+          .suppressFailureWhenFinished()
+          .onFailure((file, exc) -> LOG.debug("Failed on to remove data file {} on abort job", file.path(), exc))
+          .run(file -> io(jobContext.getJobConf()).deleteFile(file.path().toString()));
```

Comment:
Minor: `io` stores the `FileIO` after it is loaded, but it still seems odd to call the method and pass in the configuration each time. I think it would be cleaner to have `FileIO = io(jobContext.getJobConf());` at the top of this method. That keeps the handler here simpler.

Reply:
Removed the `io(Configuration configuration)` method. Easier to understand and we do not really reuse the io.
Done